### DISCUSSION TITLE
GitHub Repositories for AI Planning and Scheduling

### DISCUSSION THREAD

**Discussion by aiaustin** [w3c-cg/automated-planning#19] [Show and tell]
There are other AI planning and scheduling resources on GitHub. Please add other links with a brief description in this thread.

Let me kick this off with my own contribution and those of my AI Planning team at the University of Edinburgh. These AI planners have been built over a period of 5 decades. These include

- Traverser (1971-1972)
- Interplan (1972-1974)
- Nonlin (1974-1982)
- O-Plan (1983-1998)
- I-Plan/I-X (2000-2010)

This collection of resources was compiled as part of an effort to preserve Edinburgh AI planning materials along with the creation of an AI Planning MOOC. The source code and domain description examples are included in the repository, which was archived in the GitHub Arctic Code Vault project in 2020.

[https://github.com/aiaustin/planners](https://github.com/aiaustin/planners)

[https://open.ed.ac.uk/artificial-intelligence-planning/](https://open.ed.ac.uk/artificial-intelligence-planning/)

[https://blog.inf.ed.ac.uk/atate/2020/01/24/edinburgh-ai-planners-on-github/](https://blog.inf.ed.ac.uk/atate/2020/01/24/edinburgh-ai-planners-on-github/ )

**Comment by AdamSobieski** [comment]
Thank you. I started a group [wiki timeline of software systems](https://github.com/w3c-cg/planning/wiki/Historical-Timeline-(Software)) with that information. We also have a new [wiki timeline of languages](https://github.com/w3c-cg/planning/wiki/Historical-Timeline-(Languages)).

I can add Deviser III ([link](https://ntrs.nasa.gov/citations/19850047517), 1985) and Excalibur ([link](https://www.sciencedirect.com/science/article/abs/pii/000437029390044C), 1993) to the [wiki timeline of software systems](https://github.com/w3c-cg/planning/wiki/Historical-Timeline-(Software)), upcoming, assuming that those years are correct, 1985 and 1993?

**Reply by aiaustin** [reply]
Brian Drabble's Excalibur was created for his PhD which was completed in 1988.  The 1993 paper was an AI Journal write up quite a bit later. AI Journal in those days had long lead times.  For Brian's PhD thesis citation see https://www.aiai.ed.ac.uk/project/excalibur/

Steve Vere's Deviser was created at NASA JPL and written up in AAAI's AI Magazine by 1983. See http://www.aaai.org/ojs/index.php/aimagazine/article/download/419/355
